File: implementation/scipamato/scipamato-common-entity/src/test/java/ch/difty/scipamato/common/entity/ScipamatoEntityTest.java
package ch.difty.scipamato.common.entity;

import static ch.difty.scipamato.common.entity.ScipamatoEntity.ScipamatoEntityFields.CREATED;
import static ch.difty.scipamato.common.entity.ScipamatoEntity.ScipamatoEntityFields.MODIFIED;
import static org.assertj.core.api.Assertions.assertThat;

import java.time.LocalDateTime;

import nl.jqno.equalsverifier.EqualsVerifier;
import nl.jqno.equalsverifier.Warning;
import org.junit.Before;
import org.junit.Test;

public class ScipamatoEntityTest {

    private static final int VERSION = 10;

    private static final LocalDateTime CD = LocalDateTime
        .now()
        .minusDays(1);
    private static final LocalDateTime LM = LocalDateTime
        .now()
        .plusDays(1);

    private final ScipamatoEntity e = new ScipamatoEntity();

    @Before
    public void setUp() {
        e.setCreated(CD);
        e.setLastModified(LM);
        e.setVersion(VERSION);
    }

    @Test
    public void get() {
        assertThat(e.getCreated()).isEqualTo(CD);
        assertThat(e.getLastModified()).isEqualTo(LM);
        assertThat(e.getVersion()).isEqualTo(VERSION);
    }

    @Test
    public void testingToString() {
        assertThat(e.toString()).isEqualTo(
            "ScipamatoEntity[created=" + CD + ",lastModified=" + LM + ",version=" + VERSION + "]");
    }

    @Test
    public void equals() {
        EqualsVerifier
            .forClass(ScipamatoEntity.class)
            .withIgnoredFields(CREATED.getName(), MODIFIED.getName())
            .suppress(Warning.STRICT_INHERITANCE, Warning.NONFINAL_FIELDS)
            .verify();
    }

}

File: common/common-entity/src/test/kotlin/ch/difty/scipamato/common/entity/ScipamatoEntityTest.kt
package ch.difty.scipamato.common.entity

import ch.difty.scipamato.common.entity.ScipamatoEntity.ScipamatoEntityFields.CREATED
import ch.difty.scipamato.common.entity.ScipamatoEntity.ScipamatoEntityFields.MODIFIED
import nl.jqno.equalsverifier.EqualsVerifier
import org.amshove.kluent.shouldBeEqualTo
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.time.LocalDateTime

class ScipamatoEntityTest {

    private val e = ScipamatoEntity()

    @BeforeEach
    fun setUp() {
        e.created = CD
        e.lastModified = LM
        e.version = VERSION
    }

    @Test
    fun get() {
        e.created shouldBeEqualTo CD
        e.lastModified shouldBeEqualTo LM
        e.version shouldBeEqualTo VERSION
    }

    @Test
    fun testingToString() {
        e.toString() shouldBeEqualTo "ScipamatoEntity[created=$CD,lastModified=$LM,version=$VERSION]"
    }

    @Test
    fun equalsScipamatoEntity() {
        EqualsVerifier.simple()
            .forClass(ScipamatoEntity::class.java)
            .usingGetClass()
            .withIgnoredFields(CREATED.fieldName, MODIFIED.fieldName)
            .verify()
    }

    companion object {
        private const val VERSION = 10

        private val CD = LocalDateTime.now().minusDays(1)
        private val LM = LocalDateTime.now().plusDays(1)
    }
}
